### PR TITLE
Improve AWS/Terraform documentation

### PR DESCRIPTION
**Pull request by alicethorne-ab**
Initial rewrite of the AWS/Terraform documentation. My focus was on making it more concise and reducing the amount of jargon used, as well as moving any universally-applicable recommendations to the new PREPARING.md document.

### CODE REVIEW

File: aws-terraform/README.md [hunk 1]
```diff
@@ -38,15 +38,16 @@ For most installations, one `t3.micro` size instance will be adequate to serve I
...
+- `/module_scim_app/` - deploys the following AWS resources and their dependencies: Application Load Balancer (ALB), Auto Scaling Groups (ASG), Identity and Access Management (IAM), instance and load balancer security groups, and a public DNS record.
+- `/module_scim_infra/` - deploys network infrastructure: Virtual Private Cloud, networking (route tables, etc), Route 53, Internet Gateways, NAT Gateways, and their dependencies.
+- `/deploy/{environment}` - contains the variable files which you will need to copy and modify to suit your needs.
+
+Specific files you should review and edit include:
+
+- `variables.tf` - sets global variables for the specific deployment and provider configuration. This file already has default values for some of the variables, but it is recommended to review all of them.
+- `providers.tf` - AWS and Terraform provider configuration.
+- `main.tf` - invokes required modules and sets module specific variables.
+- `output.tf` - prints out some of the resources and values.
```
Comment (grellyd) [inline aws-terraform/README.md:51]: Eyeing the diff between the old version and your new version, I think you have nicely clarified what needs to be edited, but have also lost the nice description of the folder structure. 

I like the short heirarchical descriptions in the old version, and your new section of exactly what to edit.

Perhaps a combo of the two? Thoughts?

Comment (alicethorne-ab) [reply aws-terraform/README.md:53]: Good idea. I went ahead and made the structure example more similar to the old style, while retaining readability.

File: PREPARATION.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ There are a few common issues that pop up when deploying the SCIM Bridge.
 * You should **never** need to log into this Provision Manager account manually. Please refrain from doing so.
 * Do not attempt to perform a provisioning sync until the setup has been completed.
 * Once set up, your Identity Provider becomes the _authoritative source_ of information for your 1Password accounts. With Provisioning enabled, you _cannot edit user details manually_, and must do so through your Identity Provider.
-
+* The 1Password SCIM Bridge is not considered a service that requires high-availability. Unless you have a specific need to do so, you should only need a single instance of the service operating at a time.
```
Comment (grellyd) [inline PREPARATION.md:60]: Reading this again, I would remove the `specific need to do so` part. Running multiple SCIM Bridges is explicitly unsupported and not a case we test against.

File: aws-terraform/README.md [hunk 3]
```diff
@@ -1,104 +1,113 @@
...
 
-## General Recommendations on deploying SCIM Bridge to public cloud or private infrastructure
 
-* SCIM requests must be secured via TLS using an API gateway or load balancer (transport security).
-* Anonymous access is not supported. A bearer token and session file is required for authentication with the SCIM Bridge and the 1Password service.
-* The session file is an encrypted file containing privileged 1Password account credentials. This file is created before the bridge application is deployed. The bearer token combined with the encrypted session file represent an authentication mechanism and must be treated extremely seriously in terms of security.
+## Preparing
 
-## Prepare your 1Password Account
+This is only an example of how you could deploy the 1Password SCIM Bridge to your existing AWS infrastructure. Utilizing the full suite of AWS/Terraform infrastructure is considered an advanced form of deployment, so feel free to modify it to suit your needs.
```
Comment (grellyd) [inline aws-terraform/README.md:8]: > Utilizing the full suite of AWS/Terraform infrastructure is considered an advanced form of deployment, so feel free to modify it to suit your needs.

Minor quibble: I'm not a fan of this phrasing. Perhaps something like: "As an advanced form of deployment, the terraform deployment can be integrated into your existing infrastructure"

Up to you.

File: aws-terraform/README.md [hunk 4]
```diff
@@ -1,104 +1,113 @@
...
 
-## Logs 
-Endpoint writes to STDOUT, thus logs can be processed in a preferred way, for example, send to a remote log collector using rsyslog. A systemd unit file configuration can be used to set a specific _SyslogIdentifier_ in order to filter service logs.
+### Instance Size
 
-## DNS
-It is recommended to deploy DNS across the infrastructure to reference cache cluster/nodes and bridge endpoint by FQDN.
+For most installations, one `t3.micro` size instance will be adequate to serve Identity Provider requests. We recommend enabling auto-scaling, but it is not strictly necessary.
```
Comment (grellyd) [inline aws-terraform/README.md:26]: This reference to the size could benefit from a link to the variable in the example

https://github.com/1Password/scim-examples/blob/2c415c731d0b3924c4d247131ec2ae597eafe0f5/aws-terraform/terraform/deploy/example_env/variables.tf#L96

File: aws-terraform/README.md [hunk 5]
```diff
@@ -1,104 +1,113 @@
...
 
-5. Update/Create OP-SCIM endpoint configuration in Azure AD or Okta (IdP) using ALB URL as the endpoint and generated [_Bearer Token_](#Prepare-your-1Password-Account).
+6. Update/Create 1Password SCIM configuration in your Identity Provider using your generated [bearer token](https://github.com/1Password/scim-examples/tree/master/PREPARING.md) and newly-created subdomain.
+
+
+## Advanced
+
+### Logging
+
+All logs go to `syslog` (AWS EC2 instance OS). You can use the AWS System Manager to view logs for your instance.
```
Comment (grellyd) [inline aws-terraform/README.md:94]: Perhaps reference the actual file location in `/var/log/syslog`

File: aws-terraform/README.md [hunk 6]
```diff
@@ -1,104 +1,113 @@
...
 
-6. Application logs are available with the service start up. All logs go to `syslog` (AWS EC2 instance OS). You may consider configuring a "bastion" host or AWS System Manager or assign Public IP address to your instance, in order to be able to login to the deployed EC2 instance to view system logs.  
 
+### Upgrading and Redeployment
 
-## Deprovision (destroying) and Re-deploying
+You can destroy and redeploy the instance whenever you feel the need to. No permanent data is stored within the SCIM bridge. This is useful for upgrading your SCIM bridge with important bugfix or feature releases.
```
Comment (grellyd) [inline aws-terraform/README.md:99]: The only permanent data is stored in the secret manager you reference below. It's a slight contradiction. I would reword slightly.
